### PR TITLE
Re-enable vertical green line in raw.plot

### PR DESCRIPTION
**Pull request by cbrnr**
This PR re-enables the vertical green line when you left-click inside the plot. I've disabled marking bad channels by clicking on a trace, because this never worked reliably. Instead, bad channels can (still) be marked by clicking on the label. Right-clicking inside the plot hides the green line.

### CODE REVIEW

**Comment by cbrnr** [comment]
If the tests pass, I'll have to add this behavior to the docs. Plus update whats_new.

**Comment by larsoner** [comment]
> I've disabled marking bad channels by clicking on a trace, 

Before it used a "if you're within the y limits of the line" instead of disabling we should just use the standard `matplotlib` "if you've clicked within X pixels of the line" mode. It's a less invasive change and should make it more usable

**Comment by cbrnr** [comment]
I think it would be more usable if clicking did just one thing (setting the green line). We already have a way to mark channels by clicking on their labels. But if you insist (and give me a hint on how to implement the "click within X pixels of a line" mode) I'll change it. But maybe others who have used this can comment as well.

**Comment by larsoner** [comment]
> I think it would be more usable if clicking did just one thing (setting the green line)... But if you insist

Sometimes the signals are so bad it's not clear which channel is the problem (but one is clearly "the most problematic"), so in that case just having channel-name-clicking is not as good as also having channel-trace-clicking.

Personally I do click on traces (or at least try to). This is behavior we've had for a long time, it's likely I'm not the only one to use it, and I'd rather not risk disrupting existing patterns for little gain. Fixing the behavior / making it work better is one thing (what I'm proposing), removing it is another (what you're proposing). 

> give me a hint on how to implement the "click within X pixels of a line" mode

You can either use `line.contains(event)` or connect to the `pick_event`, which is what we do in `mne/viz/evoked.py` (see `_butterfly_onpick`).

**Comment by cbrnr** [comment]
> Sometimes the signals are so bad it's not clear which channel is the problem (but one is clearly "the most problematic"), so in that case just having channel-name-clicking is not as good as also having channel-trace-clicking.

OK, it's just that this never worked for me in such a situation. If there's just one channel going all over the place, clicking did not select the channel closest to the click location, but the channel whose location on the y-axis contained the click location.

> Personally I do click on traces (or at least try to). This is behavior we've had for a long time, it's likely I'm not the only one to use it, and I'd rather not risk disrupting existing patterns for little gain. Fixing the behavior / making it work better is one thing (what I'm proposing), removing it is another (what you're proposing).

The gain would be that the code gets simpler. But I agree if this is a pattern that people are using we should not remove it.

> You can either use `line.contains(event)` or connect to the `pick_event`, which is what we do in `mne/viz/evoked.py` (see `_butterfly_onpick`).

I'll take a look. Would this new behavior solve the issue I was talking about?

**Comment by larsoner** [comment]
> Would this new behavior solve the issue I was talking about?

Ideally in my mind after this PR if you click in white space / background, you move the green line. If you click on (or within a few px based on the `picker=2.` or whatever we pass in the `ax.plot` line call) a trace, you set it bad/good. Does that work for your case?

**Comment by cbrnr** [comment]
Yes, this (new) behavior would work.

**Comment by cbrnr** [comment]
This should now work - can you test @larsoner?

I suggest that right-clicking (anywhere) should remove the green line, what do you think? Or only right-clicking in the background (but then what should right-clicking on a signal do)? Let me know and I'll implement that as well.

**Comment by larsoner** [comment]
Yes it works great. I would let right clicking anywhere (regardless of in signal or not) remove the green line.

**Comment by mmagnuski** [comment]
What is the use case for the green line? I never found it useful - especially now once we have annotations.

**Comment by larsoner** [comment]
It can help to tell if things are actually concurrent

**Comment by mmagnuski** [comment]
Oh, right, I get it now :)

**Comment by cbrnr** [comment]
Right-clicking should now hide the green line.

**Comment by agramfort** [comment]
playing with it on a mac keyboard:

- I don't manage to remove the green bar (right click == ctrl+click on a mac)
- I see in the help Home and End to change page duration. On a mac is done with "fn + left" or "fn + right"

can this is fixed/improved here?

**Comment by cbrnr** [comment]
> I don't manage to remove the green bar (right click == ctrl+click on a mac)

Do you mean that removing the bar does not work at all for you? Or do you have to Ctrl+click to make it work? On recent macOS versions, right-clicking is enabled by default for standard mice (and you can enable right-clicking for the Apple mouse in the preferences).

> I see in the help Home and End to change page duration. On a mac is done with "fn + left" or "fn + right"

Yes, we could change this here.

**Comment by agramfort** [comment]
right-click is 2 fingers on my macbook and it does not seem to work

>

**Comment by cbrnr** [comment]
I just tested this, it works for me on my MacBook (including a two finger click on the trackpad). Did you pull in the latest commit?

**Comment by cbrnr** [comment]
And since I'm also adapting the help text, I noticed that the two columns are slightly offset:

Screen Shot 2019-03-15 at 10 29 35

However, I can't figure out why, because we call `plt.text` with both strings: https://github.com/mne-tools/mne-python/blob/fb781d5c0c693289942ba9f87db7a4cbf7ea3f83/mne/viz/utils.py#L1072

**Comment by agramfort** [comment]
I confirm that the 2 finger click does not work to remove green bar for me
but this can be investigated later.

+1 for MRG

**Comment by larsoner** [comment]
Right clicking works for me on macOS Mojave w/PyQt5 backend. @agramfort maybe you are on the Cocoa / OSX matplotlib backend?

+1 for merge after whats_new.rst update

**Comment by agramfort** [comment]
I am still using Sierra. I use Qt5Agg

**Comment by cbrnr** [comment]
The help text looks now better and uses `plt.table` (much simpler):

Screen Shot 2019-03-15 at 12 16 08

**Comment by cbrnr** [comment]
Done. On my Mac, right-clicking works with both `Qt5Agg` and `macosx` backends.

**Comment by larsoner** [comment]
Nice! If you're up for it, `set_text_props(ha='right')` on the left column might help align the items on the left with their description on the right, see loop at the end of 

https://matplotlib.org/gallery/text_labels_and_annotations/font_table_ttf_sgskip.html

File: mne/viz/utils.py [hunk 1]
```diff
@@ -321,11 +323,18 @@ def _get_help_text(params):
...
     text.append(u'- : \n')
     text.append(u'+ or = : \n')
-    text.append(u'Home : \n')
-    text.append(u'End : \n')
-    if 'fig_selection' not in params:
-        text.append(u'Page down : \n')
-        text.append(u'Page up : \n')
+    if is_mac:
+        text.append(u'fn + \u2190 : \n')
+        text.append(u'fn + \u2192 : \n')
```
Comment (larsoner) [inline mne/viz/utils.py]: Probably cleaner just to put in the chars (e.g. ←) directly. You might need to add `# -*- coding: utf-8 -*-` to the top of the file

File: mne/viz/utils.py [hunk 2]
```diff
@@ -1051,39 +1063,34 @@ def f(x, y):
...
-    plt.axis('off')
+    ax = fig_help.add_subplot(111)
+    celltext = [[c1, c2] for c1, c2 in zip(text.strip().split("\n"),
+                                           text2.strip().split("\n"))]
+    table = ax.table(cellText=celltext, loc="center", cellLoc="left")
+    table.auto_set_font_size(False)
+    table.set_fontsize(12)
+    ax.set_axis_off()
+    for _, cell in table.get_celld().items():
+        cell.set_edgecolor(None)  # remove cell borders
```
Comment (larsoner) [inline mne/viz/utils.py]: Oh you already have a loop -- so doing `key, cell` then `row, col = key` and `if key == 0: cell.set_text_props(ha='right')` should help the alignment

Comment (cbrnr) [reply mne/viz/utils.py]: I've tried that, but for whatever reason this doesn't work - the text is still left-aligned...

**Comment by larsoner** [comment]
@cbrnr pushed a commit to right justify

**Comment by cbrnr** [comment]
I saw that but thought using an internal property wasn't worth the benefit. But you're right, it looks better!

**Comment by larsoner** [comment]
Yes it's a bit of a risk, but I figure the most likely problem we'll see if they drop "support" for it is that it'll go back to left aligning

**Comment by agramfort** [comment]
old version is more readable I think. Here it's harder to see what line
maps to what on right side

>

**Comment by larsoner** [comment]
That should be fixed in the latest commit @agramfort 

Screen Shot 2019-03-15 at 07 55 21

**Comment by agramfort** [comment]
LGTM +1 for MRG

>

**Comment by larsoner** [comment]
Thanks @cbrnr

**Comment by cbrnr** [comment]
@agramfort I can sort of replicate the behavior of the green line not disappearing on right-clicking. However, I can replicate only when I create the plot from MNELAB, not directly with MNE code. Furthermore, when I debug in MNELAB, the behavior is also correct. It seems like Matplotlib doesn't refresh although I correctly set the visibility of the line to `False`. Any ideas?

**Comment by cbrnr** [comment]
@larsoner maybe you want to try if you can replicate with MNELAB?

**Comment by agramfort** [comment]
not much ideas

>

**Comment by larsoner** [comment]
Right clicking works fine for me in `mnelab`

**Comment by cbrnr** [comment]
That's weird. Given that I can't reproduce it in debug mode, there's probably not much we can do.

**Pull request merged**